core/mempool/src/tx_gossip.rs

```diff
@@ -0,0 +1,26 @@
+use primitives::types::AuthorityId;
+use primitives::chain::ChainPayload;
+use primitives::signature::Signature;
+use primitives::hash::hash_struct;
+use primitives::signature::SecretKey;
+use primitives::signature::sign;
+
+#[derive(Debug, Serialize, Deserialize, Eq, PartialEq)]
+pub struct TxGossip {
```

Comment:
Should this be PayloadGossip?
Also should gossip first send Hashs of tx/receipts and then other client retrieves content if doesn't have it?